### PR TITLE
(DOCSP-27239) [Admin API] App Logs

### PR DESCRIPTION
**Pull request by nlarew**
### Jira

- (DOCSP-27239) [Admin API] App Logs

### Staged Changes

- [Admin API > Get Logs](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/admin-logs/admin/api/v3/#operation/adminGetLogs)

### CODE REVIEW

File: source/openapi-admin-v3.yaml [hunk 1]
```diff
@@ -7102,6 +7070,590 @@ components:
...
+                    },
+                    "no_matching_role": 0
+                  }
+                }
+              }
+              properties:
+                namespaces_metrics:
+                  type: object
+                  additionalProperties:
+                    description: Metrics for a specific `database.collection` namespace
```
Comment (cbullinger) [inline source/openapi-admin-v3.yaml:7134]:
```suggestion
                    description: Metrics for a specific `database.collection` namespace.
```
for consistency

File: source/openapi-admin-v3.yaml [hunk 2]
```diff
@@ -7102,6 +7070,590 @@ components:
...
+                    type: object
+                    properties:
+                      roles:
+                        type: object
+                        additionalProperties:
+                          type: object
+                          description: Metrics for a specific role
```
Comment (cbullinger) [inline source/openapi-admin-v3.yaml:7141]:
```suggestion
                          description: Metrics for a specific role.
```

File: source/openapi-admin-v3.yaml [hunk 3]
```diff
@@ -7102,6 +7070,590 @@ components:
...
+              data_source_name:
+                type: string
+                example: "mongodb-atlas"
+                description: |
+                  The name of the forwarded log data source.
+              data_source_namespace:
+                type: string
+                example: "myApp.logs"
+                description: |
+                  The `database.collection` namespace of the forwarded log collection.
```
Comment (cbullinger) [inline source/openapi-admin-v3.yaml:7654]: should this be 
```suggestion
                  The ``database.collection`` namespace of the forwarded log collection.
```

File: source/openapi-admin-v3.yaml [hunk 4]
```diff
@@ -7102,6 +7070,590 @@ components:
...
+            - $ref: "#/components/schemas/Logs/Parts/BaseLog"
+            - $ref: "#/components/schemas/Logs/Parts/BilledMemory"
+            - $ref: "#/components/schemas/Logs/Parts/FunctionCall"
+            - $ref: "#/components/schemas/Logs/Parts/Messages"
+            - properties:
+                event_subscription_id:
+                  type: string
+                  example: "641ca0709e425c794edc7c6d"
+                  description: |
+                    The triggers's unique objectId hex.
```
Comment (cbullinger) [inline source/openapi-admin-v3.yaml:7262]:
```suggestion
                    The trigger's unique objectId hex.
```

File: source/openapi-admin-v3.yaml [hunk 5]
```diff
@@ -7102,6 +7070,590 @@ components:
...
+                event_subscription_name:
+                  type: string
+                  example: "myDatabaseTrigger"
+                  description: |
+                    The triggers's name.
```
Comment (cbullinger) [inline source/openapi-admin-v3.yaml:7267]:
```suggestion
                    The trigger's name.
```

File: source/openapi-admin-v3.yaml [hunk 6]
```diff
@@ -7102,6 +7070,590 @@ components:
...
+              example: { "name": "data/v1/insertOne", "arguments": null }
+            function_id:
+              type: string
+              example: "63866254a4e9d232454e6334"
+            function_name:
+              type: string
+              example: "myFunction"
+        BaseLog:
+          type: object
+          description: Common fields shared by all log types
```
Comment (cbullinger) [inline source/openapi-admin-v3.yaml:7203]:
```suggestion
          description: Common fields shared by all log types.
```

File: source/openapi-admin-v3.yaml [hunk 7]
```diff
@@ -2519,7 +2519,7 @@ paths:
       parameters:
         - name: co_id
           in: query
-          description: Return only log messages associated with the given request ID.
+          description: Return only log messages associated with the given request Correlation ID.
```
Comment (cbullinger) [inline source/openapi-admin-v3.yaml:2522]: why capitalized?

**Review by cbullinger — APPROVED** [review]
staged page looks good. a couple of small suggestions

File: source/openapi-admin-v3.yaml (hunk 3 above)
Comment (nlarew) [reply source/openapi-admin-v3.yaml:7654]: The OpenAPI spec uses markdown, not reStructuredText, so the single backticks are actually correct here

File: source/openapi-admin-v3.yaml (hunk 7 above)
Comment (nlarew) [reply source/openapi-admin-v3.yaml:2522]: I'm sort of using it as a proper noun here, mainly because what I've read about it also uses that style.

**Pull request merged**